### PR TITLE
Temporarily add a fix to use account state as "clean" if 'free' fails…

### PR DESCRIPTION
**Pull request by harishspqr**
**What this PR does / why we need it**:
It fixes the AWS E2E tests running in test-infra. 

```release-note
NONE
```

### CODE REVIEW

File: hack/checkout_account.py [hunk 1]
```diff
@@ -27,24 +27,29 @@
...
     if resp.status != 200:
         sys.exit("Got invalid response %d: %s" % (resp.status, resp.reason))
 
     body = json.load(resp)
     conn.close()
     print 'export BOSKOS_RESOURCE_NAME="%s";' % body['name']
     print 'export AWS_ACCESS_KEY_ID="%s";' % body['userdata']['access-key-id']
-    print 'export AWS_SECRET_ACCESS_KEY="%s";' % body['userdata']['secret-access-key']
-
+    print 'export AWS_SECRET_ACCESS_KEY="%s";' % body['userdata']['secret-access-key']
```
Comment (ncdc) [inline hack/checkout_account.py]: Does your editor remove newlines at the end of the file?

File: hack/checkout_account.py [hunk 2]
```diff
@@ -27,24 +27,29 @@
 RESOURCE_TYPE = "aws-account"
 USER = "cluster-api-provider-aws"
 
-if __name__ == "__main__":
-    conn = httplib.HTTPConnection(BOSKOS_HOST)
-
+def post_request(input_state = "free"):
```
Comment (ncdc) [inline hack/checkout_account.py]: I'd add a comment that we're working around an issue with the data in boskos and that we'll remove this code that tries both free and clean once all the data is good.

File: hack/checkout_account.py [hunk 3]
```diff
@@ -27,24 +27,29 @@
...
     })
 
     )
-    resp = conn.getresponse()
+    return conn.getresponse()
+
+if __name__ == "__main__":
+    conn = httplib.HTTPConnection(BOSKOS_HOST)
+
+    resp = post_request()
```
Comment (ncdc) [inline hack/checkout_account.py]: You'll need to pass `conn` in, won't you?

**Comment by detiber** [comment]
We should probably hold off on this to see if @BenTheElder is able to fix the state for us.

File: hack/checkout_account.py (hunk 1 above)
Comment (harishspqr) [reply hack/checkout_account.py]: Not sure. Will add back.

File: hack/checkout_account.py (hunk 3 above)
Comment (harishspqr) [reply hack/checkout_account.py]: absolutely!

**Pull request merged**